src/types/closure.md

```diff
@@ -178,3 +293,258 @@ Because captures are often by reference, the following general rules arise:
 [`Sync`]: ../special-types-and-traits.md#sync
 [closure expression]: ../expressions/closure-expr.md
 [derived]: ../attributes/derive.md
+
+## Drop Order
+
+If a closure captures a field of a composite types such as structs, tuples, and enums by value, the field's lifetime would now be tied to the closure. As a result, it is possible for disjoint fields of a composite types to be dropped at different times.
+
+```rust
+{
+    let tuple =
+      (String::from("foo"), String::from("bar")); // --+
+    { //                                               |
+        let c = || { // ----------------------------+  |
+            // tuple.0 is captured into the closure |  |
+            drop(tuple.0); //                       |  |
+        }; //                                       |  |
+    } // 'c' and 'tuple.0' dropped here ------------+  |
+} // tuple.1 dropped here -----------------------------+
+```
+
+
+## Overall Capture analysis algorithm
+
+* Input:
+    * Analyzing the closure C yields a mapping of `Place -> Mode` that are accessed
+    * Access mode is `ref`, `ref uniq`, `ref mut`, or `by-value` (ordered least to max)
```

Comment:
I'm still a bit unclear on the best way to incorporate ref uniq here -- my sense is that the results that come in initially should never have ref uniq, but it gets determined later, whenever we do a truncation through a `*` of an `&mut` (as we talked about).